src/rpc_client/responses.rs

```diff
@@ -0,0 +1,93 @@
+use crate::rpc_client::checked_amount::CheckedAmountOf;
+use crate::rpc_client::eth_rpc::{Hash, HttpResponsePayload, LogEntry, ResponseTransform};
+use crate::rpc_client::numeric::{BlockNumber, GasAmount, WeiPerGas};
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct TransactionReceipt {
+    /// The hash of the block containing the transaction.
+    #[serde(rename = "blockHash")]
+    pub block_hash: Hash,
+
+    /// The number of the block containing the transaction.
+    #[serde(rename = "blockNumber")]
+    pub block_number: BlockNumber,
+
+    /// The total base charge plus tip paid for each unit of gas
```

Comment:
OCD alert! Some comments here end with a period, others don't.   :slightly_smiling_face:

Reply:
Great PR! Thanks a lot! I'll approve it now since I only spotted a few (very) minor issues.